controller_manager/scripts/spawner.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python3
+# Copyright 2021 PAL Robotics S.L.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import subprocess
+import sys
+import time
+
+import rclpy
+from rclpy.duration import Duration
+from rclpy.node import Node
+
+
+def is_controller_loaded(controller_manager_name, controller_name):
+    ret = subprocess.run(['ros2', 'control', 'list_controllers',
+                          '--controller-manager', controller_manager_name], capture_output=True,
+                         encoding='utf8')
```

Comment:
I am not a big fan of spawning a subprocess here. This adds another dependency here (`ros2control`) which is unnecessary. Also, wouldn't you create a circular dependency here?

Why not directly call the controller manager service?

Reply:
The dependency is `ros2controlcli` which would not make a circular dependency.

I went for the subprocess approach because it was faster, but also to avoid duplicating the `ros2controlcli` code and also because loading parameter files is not trivial, there's no standard api for it (https://github.com/ros2/rclpy/issues/671) so I feel like we'd still keep using a subprocess for `ros2 param load` to avoid duplicating all that.

@destogl also mentioned using the services directly, my proposal was to use this to have something ready quick, but it turns out that we're still one month away from the sync that includes `ros2 param load` so we may have time.

On the other hand, I'm going to disappear for some months any time now (paternity leave) and I'd like to not find this waiting here when I'm back, so if I can't do it the next few days I'd ask to either someone give it a shot to change it, or accept as it is and improve it later.

Reply:
https://github.com/ros-controls/ros2_control/issues/344